zproject/prod_settings_template.py

```diff
@@ -764,6 +764,15 @@
 ## Controls whether Zulip will rate-limit user requests.
 # RATE_LIMITING = True
 
+## Entries in this dictionary will override Zulip's default rate limits,
+## which are hard-coded in the codebase in RATE_LIMITING_RULES. Rules
+## for domains which are not explicitly overridden here will be as default.
+# OVERRIDDEN_RATE_LIMITING_RULES = {
+#     "api_by_ip": [
+#         (60, 100),
+#     ],
+# }
```

Comment:
Could we set `DEFAULT_RATE_LIMITING_RULES` in `default_settings.py`, let the user set `RATE_LIMITING_RULES` in `settings.py`, and in `computed_settings.py` do:
```py
RATE_LIMITING_RULES = {**DEFAULT_RATE_LIMITING_RULES, **RATE_LIMITING_RULES}
# or
RATE_LIMITING_RULES = {k: RATE_LIMITING_RULES.get(k, DEFAULT_RATE_LIMITING_RULES[k]) for k in DEFAULT_RATE_LIMITING_RULES.keys()}
```

It seems a bit odd to have the `OVERRIDDEN` in the name, since pretty much all of `settings.py` is overriding _some_ default config.

Reply:
Yeah, I like that idea!

Reply:
Okay, makes sense, tweaked! The idea for the odd "OVERRIDEN" in the name was, because unlike a typical setting, which overrides "the whole thing" in a sense, this actually does not override the full `RATE_LIMITING_RULES` dict, but rather does it on a rule-by-rule basis. But it seems fine to me to drop `OVERRIDEN`.